src/Configuration/src/ConfigServer/ConfigServerConfigurationProvider.cs

```diff
@@ -735,14 +734,14 @@ internal bool IsDiscoveryFirstEnabled()
     /// </returns>
     internal HttpClient CreateHttpClient(ConfigServerClientOptions options)
     {
-        ArgumentGuard.NotNull(options);
+        ArgumentNullException.ThrowIfNull(options);
         ObjectDisposedException.ThrowIf(_httpClientHandler == null, this);
 
         var clientCertificateConfigurer = new ClientCertificateHttpClientHandlerConfigurer(OptionsMonitorWrapper.Create(options.ClientCertificate));
-        clientCertificateConfigurer.Configure(_httpClientHandler);
+        clientCertificateConfigurer.Configure("ConfigServer", _httpClientHandler);
 
         var validateCertificatesHandler = new ValidateCertificatesHttpClientHandlerConfigurer<ConfigServerClientOptions>(OptionsMonitorWrapper.Create(options));
-        validateCertificatesHandler.Configure(_httpClientHandler);
+        validateCertificatesHandler.Configure(string.Empty, _httpClientHandler);
```

Comment:
Can/should we be using `Options.DefaultName` here? It looks like there is currently a naming conflict, but we could rename the property

Reply:
You're right, I mistakenly assumed the error was caused by not having a dependency on the options library. I'll rename the property.